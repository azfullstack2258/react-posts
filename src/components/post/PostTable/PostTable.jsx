import React from 'react';
import PropTypes from 'prop-types';

import PostTableHeader from '../PostTableHeader';
import PostTableRow from '../PostTableRow';

const PostTable = ({ posts }) => {
  return (
    <div className="post-table">
      <div className="post-table__body">
        <PostTableHeader />
        {Array.isArray(posts) ? (
          posts.map(post => <PostTableRow key={post.id} data={post} />)
        ) : (
          <div className="post-table__body--empty">No posts</div>
        )}
      </div>
    </div>
  );
};

PostTable.propTypes = {
  posts: PropTypes.arrayOf(PropTypes.object),
};

PostTable.defaultProps = {
  posts: []
};

export default PostTable;
