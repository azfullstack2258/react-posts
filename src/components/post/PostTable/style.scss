.post-table__body {
  overflow-x: auto;
}
