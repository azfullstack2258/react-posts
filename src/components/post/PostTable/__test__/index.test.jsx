import React from 'react';
import { render } from '@testing-library/react';

import PostTable from '../PostTable';

describe('<PostTable />', () => {
  it('should render a prop', () => {
    const { container } = render(
      <PostTable />
    );

    expect(container).toMatchSnapshot();
  })
});
