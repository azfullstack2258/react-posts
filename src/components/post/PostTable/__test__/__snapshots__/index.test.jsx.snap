// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTable /> should render a prop 1`] = `
<div>
  <div
    class="post-table"
  >
    <div
      class="post-table__body"
    >
      <div
        class="post-table__header"
      >
        <div
          class="post-table__row-cell id"
        >
          ID
        </div>
        <div
          class="post-table__row-cell property"
        >
          Property
        </div>
        <div
          class="post-table__row-cell user"
        >
          User
        </div>
        <div
          class="post-table__row-cell network"
        >
          Network
        </div>
        <div
          class="post-table__row-cell date"
        >
          Date
        </div>
        <div
          class="post-table__row-cell image"
        >
          Image
        </div>
        <div
          class="post-table__row-cell content"
        >
          Content
        </div>
      </div>
    </div>
  </div>
</div>
`;
